R/gh_token.R

```diff
@@ -164,21 +180,162 @@ gh_auth <- function(token) {
 #' # A custom one
 #' slugify_url("https://github.acme.com")
 
-slugify_url <- function(url) {
+slugify_url <- function(url) {           # https://jane@github.uni.edu/api/v3
+  url <- get_baseurl(url)                # https://jane@github.uni.edu
+  url <- normalize_host(url)
+  x2 <- sub("^.*://([^/]*@)?", "", url)  # github.uni.edu
+  x3 <- gsub("[.]+", "_", x2)            # github_uni_edu
+  x4 <- gsub("[^-a-zA-Z0-9_]", "", x3)
+  toupper(x4)                            # GITHUB_UNI_EDU
+}
+
+get_baseurl <- function(url) {               # https://github.uni.edu/api/v3/
   if (!any(grepl("^https?://", url))) {
     stop("Only works with HTTP(S) protocols")
   }
-  x2 <- sub("^.*://([^/]*@)?", "", url)
-  x3 <- sub("/+$", "", x2)
-  x4 <- gsub("[./]+", "_", x3)
-  x5 <- gsub("[^-a-zA-Z0-9_]", "", x4)
-  toupper(x5)
+  prot <- sub("^(https?://).*$", "\\1", url) # https://
+  rest <- sub("^https?://(.*)$", "\\1", url) #         github.uni.edu/api/v3/
+  host <- sub("/.*$", "", rest)              #         github.uni.edu
+  paste0(prot, host)                         # https://github.uni.edu
+}
+
+normalize_host <- function(x) {
+  sub("api.github.com", "github.com", x)
+}
+
+get_hosturl <- function(url) {
+  url <- get_baseurl(url)
+  normalize_host(url)
+}
+
+get_apiurl <- function(url) {
```

Comment:
Since we accept "https://github.com" and "https://github.ubc.ca" instead of insisting on "https://api.github.com" and "https://github.ubc.ca/api/v3", we need this function to ensure we have the true API root url.